pdns/dnsdist-lua.cc

```diff
@@ -1647,6 +1647,18 @@ static void setupLuaConfig(bool client, bool configCheck)
       }
     });
 
+  g_lua.writeFunction("setWeightedBalancingFactor", [](double factor) {
+      setLuaSideEffect();
+      if (factor >= 0) {
```

Comment:
we could get strange behavior in case of a factor that is set between 0 and 1 no ?

Reply:
Done!